README.md

```diff
@@ -23,85 +175,121 @@ Copy and paste into your Terraform configuration, insert the variables, and run
+
+`graphdb_java_options = "-XX:+PrintCommandLineFlags"`
+
+**Custom GraphDB VM Image**
+`ami_id = "ami-1241215424634"`
 
-## Modules
+**Purge Protection**
+Resources that support purge protection have them enabled by default. You can override the default configurations with
+the following variables:
+`prevent_resource_deletion = false`
 
-#### [Backup](modules/backup/README.md)
-This Terraform module creates an Amazon S3 bucket for storing backups.
+**Backup**
 
-#### [Config](modules/config/README.md)
-This Terraform module creates AWS Systems Manager (SSM) parameters for managing sensitive data and configuration settings used by GraphDB in an AWS environment.
+To enable deployment of the backup module, you need to enable the following flag:
 
-#### [DNS](modules/dns/README.md)
-This Terraform module creates a private hosted zone in Amazon Route 53 for DNS resolution in your Virtual Private Cloud (VPC).
+`deploy_backup = true`
 
-#### [IAM](modules/iam/README.md)
-This Terraform module creates an AWS Identity and Access Management (IAM) role and an instance profile that can be used for EC2 instances. It also supports the option to set a permissions boundary on the IAM role.
+**Monitoring**
 
-#### [Load Balancer](modules/load_balancer/README.md)
-This Terraform module sets up an AWS Elastic Load Balancer (Network Load Balancer) with optional TLS listeners. The module is designed to be flexible and customizable by accepting various input variables to tailor the NLB configuration to your specific requirements.
+To enable deployment of the monitoring module, you need to enable the following flag:
 
-#### [User Data](modules/user_data/README.md)
-This Terraform module configures an AWS EC2 instance for running GraphDB with various optional parameters and user-supplied userdata.
+`deploy_monitoring = true`
 
-#### [VM](modules/vm/README.md)
-This Terraform module configures IAM roles, EC2 instances, and associated security groups for running GraphDB with optional parameters and customization.
+**Providing a TLS certificate**
 
-#### [MONITORING](modules/monitoring/README.md)
-This Terraform module configures Cloudwatch Dashboard, Cloudwatch Alarms, Metric filters and log group.
+`lb_tls_certificate_arn = "arn:aws:acm:us-east-1:123456789012:certificate/12345678-1234-1234-1234-123456789012` #
+Example ARN
+
+
+
+***Important Variables (Inputs)***
+
+The following are the important variables you should configure when using this module:
+
+- `aws_region`: The region in which GraphDB will be deployed.
+
+- `vpc_id`: The ID of the Virtual Private Cloud (VPC) where you want to create the AWS GraphDB cluster. This is a
+  required variable.
+
+- `ec2_instance_type`: The instance type for the GDB cluster nodes. This should match your performance and cost
+  requirements.
+
+- `node_count`: The number of instances in the cluster. Recommended is 3, 5 or 7 in order to have consensus according to
+  the [Raft algorithm](https://raft.github.io/).
+
```

Comment:
license ?
version ? 
password? there is no mention that if the password is not set it will be auto generated.